### PR TITLE
Created String Palindrome program in Java

### PR DESCRIPTION
**Pull request by iamvs-2002**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
Yes
(Write your answer here.)

### Description
I have implemented a java program to check if a string entered by the user is a palindrome or not.
(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#1925 
(Write your answer here.)

### CODE REVIEW

**Comment by srimanikantapalakollu** [comment]
@iamvs-2002 Your branch has some conflicts, please check it at your end make the PR appropriately.

**Review by srimanikantapalakollu — CHANGES_REQUESTED** [review]
Along with Conflicts, Your code will take O(n) time complexity where you can solve it in O(n/2) time only. please have a look at it and modify it appropriately.

**Comment by iamvs-2002** [comment]
> Along with Conflicts, Your code will take O(n) time complexity where you can solve it in O(n/2) time only. please have a look at it and modify it appropriately.

Oh, I see. Thanks for pointing that out!

File: Java/other/Palindrome.java [hunk 1]
```diff
@@ -4,6 +4,7 @@
 A program to check whether a string is palindrome or not.
 A plaindrome string is a string which is same to its reverse.
 eg. 1234321, radar etc.
+Time Complexity = O(n/2)
```
Comment (atarax665) [inline Java/other/Palindrome.java]: Add space complexity too.

File: Java/other/Palindrome.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    //method to check if the string is a palindrome or not
+    //method returns true if the string is a palindrome else it returns false
+    private static boolean ispalindrome(String string_1) 
+    {
+        //storing the last index of the string in n
+        int n =string_1.length() - 1;
+        //comparing the characters of the string
+        //first to the end and so on
+        for (int i = 0; i < (n)/2; i++)
```
Comment (atarax665) [inline Java/other/Palindrome.java]:
```suggestion
        for (int i = 0; i < (n) / 2; i++) 
```

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Review by atarax665 — APPROVED** [review]
LGTM :+1:

**Comment by srimanikantapalakollu** [comment]
@iamvs-2002 is this part of any open source program?

**Comment by iamvs-2002** [comment]
> @iamvs-2002 is this part of any open source program?

Yes, DWOC

**Pull request merged**